benchopt/utils/misc.py

```diff
@@ -84,3 +85,23 @@ def get_benchopt_requirement(pytest=False):
             )
 
     return req, False
+
+
+def OSSpecificNamedTempFile(dir=None, mode='w+b', prefix=None, suffix=None):
+    if sys.platform != 'win32':
+        return tempfile.NamedTemporaryFile(dir=dir, mode=mode, prefix=prefix,
+                                           suffix=suffix)
+
+    else:
+        required_version = (3, 12)
+        current_version = sys.version_info
+
+        if current_version < required_version:
+            raise EnvironmentError("Your current Python version is "
```

Comment:
you can indent the code by breaking line after the opening parenthesis to have longer lines and thus fewer lines